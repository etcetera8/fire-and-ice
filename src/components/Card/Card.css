.display-card {
  border: 1px solid black;
}