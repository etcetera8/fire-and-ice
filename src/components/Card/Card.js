import React, { Component } from 'react';
import './Card.css'

export class Card extends Component {
  constructor(props){
    super(props)
    this.state = {
      clicked: false
    }
  }

  clicked = () => {
    console.log(this.state);
    if (this.state.clicked === false) {
      this.setState({clicked: true})
    } else if (this.state.clicked === true) {
      this.setState({clicked: false})
      }
    }
  
  
  render () {
    const {name, founded, seats, titles, coatOfArms, ancestralWeapons, words, swornMembers} = this.props;
    return (
      <section className='Card' onClick={this.clicked}>
        <span className="house-info">{name}</span>
        <span className="house-info">Founded: {founded}</span>
        <span className="house-info">Seats: {seats}</span>
        <span className="house-info">Titles: {titles}</span>
        <span className="house-info">Coat of Arms: "{coatOfArms}"</span>
        <span className="house-info">Ancestral Weapons: {ancestralWeapons}</span>
        <span className="house-info">Words: {words}</span>
        {
          this.state.clicked && 
          <span>{swornMembers}</span>
        }
      </section>
    )
  }
}



export default Card;