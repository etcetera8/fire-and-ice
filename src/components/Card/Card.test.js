import React from 'react';
import { shallow } from 'enzyme';
import {Card} from './Card';
const mockHouse = {
  name:"house",
  founded: 1,
  seats: 2,
  titles: 3,
  coatOfArms: "tvshow",
  ancestralWeapons: "sword",
  words: '',
  swornMembers: ["neb"]
}

describe('Card', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Card house={mockHouse}/>)
    expect(wrapper).toMatchSnapshot();
  })

  it('should have a default state of not clicked', () => {
    const wrapper = shallow(<Card house={mockHouse}/>)
    expect(wrapper.state().clicked).toEqual(false)
  })

  it('when clicked its state should be switched', () => {
    const wrapper = shallow(<Card house={mockHouse}/>)
    const inst = wrapper.instance();
    inst.clicked();
    expect(wrapper.state().clicked).toEqual(true)
  })

  it('should create an jsx array of swornMembers', () => {
    const wrapper = shallow(<Card house={mockHouse}/>)
    const inst = wrapper.instance();
    const results = inst.createMembers([{name: 'neb', gender:'male', died: false}])
    expect(results.length).toEqual(1);
  })
})