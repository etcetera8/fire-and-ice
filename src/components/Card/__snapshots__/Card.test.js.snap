// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    house={
        Object {
            "ancestralWeapons": "sword",
            "coatOfArms": "tvshow",
            "founded": 1,
            "name": "house",
            "seats": 2,
            "swornMembers": Array [
              "neb",
            ],
            "titles": 3,
            "words": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="house-info"
>
          house
</span>,
        <span
          className="house-info"
>
          Founded: 
          1
</span>,
        <span
          className="house-info"
>
          Seats: 
          2
</span>,
        <span
          className="house-info"
>
          Titles: 
          3
</span>,
        <span
          className="house-info"
>
          Coat of Arms: "
          tvshow
          "
</span>,
        <span
          className="house-info"
>
          Ancestral Weapons: 
          sword
</span>,
        <span
          className="house-info"
>
          Words: 
          
</span>,
        false,
      ],
      "className": "Card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "house",
          "className": "house-info",
        },
        "ref": null,
        "rendered": "house",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Founded: ",
            1,
          ],
          "className": "house-info",
        },
        "ref": null,
        "rendered": Array [
          "Founded: ",
          1,
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Seats: ",
            2,
          ],
          "className": "house-info",
        },
        "ref": null,
        "rendered": Array [
          "Seats: ",
          2,
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Titles: ",
            3,
          ],
          "className": "house-info",
        },
        "ref": null,
        "rendered": Array [
          "Titles: ",
          3,
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Coat of Arms: \\"",
            "tvshow",
            "\\"",
          ],
          "className": "house-info",
        },
        "ref": null,
        "rendered": Array [
          "Coat of Arms: \\"",
          "tvshow",
          "\\"",
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Ancestral Weapons: ",
            "sword",
          ],
          "className": "house-info",
        },
        "ref": null,
        "rendered": Array [
          "Ancestral Weapons: ",
          "sword",
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Words: ",
            "",
          ],
          "className": "house-info",
        },
        "ref": null,
        "rendered": Array [
          "Words: ",
          "",
        ],
        "type": "span",
      },
      false,
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="house-info"
>
            house
</span>,
          <span
            className="house-info"
>
            Founded: 
            1
</span>,
          <span
            className="house-info"
>
            Seats: 
            2
</span>,
          <span
            className="house-info"
>
            Titles: 
            3
</span>,
          <span
            className="house-info"
>
            Coat of Arms: "
            tvshow
            "
</span>,
          <span
            className="house-info"
>
            Ancestral Weapons: 
            sword
</span>,
          <span
            className="house-info"
>
            Words: 
            
</span>,
          false,
        ],
        "className": "Card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "house",
            "className": "house-info",
          },
          "ref": null,
          "rendered": "house",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Founded: ",
              1,
            ],
            "className": "house-info",
          },
          "ref": null,
          "rendered": Array [
            "Founded: ",
            1,
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Seats: ",
              2,
            ],
            "className": "house-info",
          },
          "ref": null,
          "rendered": Array [
            "Seats: ",
            2,
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Titles: ",
              3,
            ],
            "className": "house-info",
          },
          "ref": null,
          "rendered": Array [
            "Titles: ",
            3,
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Coat of Arms: \\"",
              "tvshow",
              "\\"",
            ],
            "className": "house-info",
          },
          "ref": null,
          "rendered": Array [
            "Coat of Arms: \\"",
            "tvshow",
            "\\"",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Ancestral Weapons: ",
              "sword",
            ],
            "className": "house-info",
          },
          "ref": null,
          "rendered": Array [
            "Ancestral Weapons: ",
            "sword",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Words: ",
              "",
            ],
            "className": "house-info",
          },
          "ref": null,
          "rendered": Array [
            "Words: ",
            "",
          ],
          "type": "span",
        },
        false,
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
